### PR TITLE
elearn eeclass oauth + eeclass announcement

### PR DESCRIPTION
**Pull request by waltsai2483**

### CODE REVIEW

**Comment by ImJustChew** [comment]
@waltsai2483 can you merge this branch with `main`.

**Comment by waltsai2483** [comment]
@ImJustChew done

**Comment by ImJustChew** [comment]
Once you're done rebasing with upstream( merge with main), just leave it here and I'll review your code. Keep the pull request open.

**Comment by waltsai2483** [comment]
@ImJustChew I have done merging with main, but there's a bug that causes attachments in eeclass announcements expire and I'm fixing it
I'll notice you once it's finished

**Comment by ImJustChew** [comment]
No problem, just change this PR into a draft PR.

File: src/const/apps.ts
Comment (ImJustChew) [inline src/const/apps.ts:26]: Merge Error, clubs  should be maintained

**Review by ImJustChew — CHANGES_REQUESTED** [review]
Please fix the merge error, but apart from that, everything looks very good. Thanks @waltsai2483. @Joshimello will help you improve the UI. Will merge to `elearn-oauth` branch once done.

**Pull request merged**
